gh-92886: Replace assertion statements in `handlers.BaseHandler` to support running with optimizations (`-O`) (#93231)
python/cpython: Misc/NEWS.d/next/Library/2022-05-25-22-09-38.gh-issue-92886.ylwDSc.rst

> Replace ``assert`` statements with ``raise AssertionError()`` in :class:`BaseHandler` so that behaviour is maintained and tests in ``test_wsgiref.py`` pass when running with optimizations ``(-O)``.

Reviewer: ```suggestion
Replace ``assert`` statements with ``raise AssertionError()`` in :class:`~wsgiref.BaseHandler` so that the tested behaviour is maintained running with optimizations ``(-O)``.
```
